src/Symfony/Component/Asset/Package.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset;
+
+use Symfony\Component\Asset\Context\ContextInterface;
+use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+/**
+ * Basic package that adds a version to asset URLs.
+ *
+ * @author Kris Wallsmith <kris@symfony.com>
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class Package implements PackageInterface
+{
+    private $versionStrategy;
+    private $context;
+
+    public function __construct(VersionStrategyInterface $versionStrategy)
+    {
+        $this->versionStrategy = $versionStrategy;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getVersion($path)
+    {
+        return $this->versionStrategy ? $this->versionStrategy->getVersion($path) : '';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getUrl($path)
+    {
+        if ($this->isAbsoluteUrl($path)) {
+            return $path;
+        }
```

Comment:
We have got some feature request about being able to apply versions on absolute URLs too when using the query string for cache busting (the main use case being image links generated by LiipImagineBundle IIRC). I cannot find the issue again though (searching for ``version`` in my github email notifications brings me all issues talking about libraries versions, making it hard).
It might be a good idea to try solving this case in the new componentt